### PR TITLE
grafx2: new port, version 2.5  (new Portfile)

### PR DESCRIPTION
**Pull request by miniupnp**
Signed-off-by: Thomas Bernard <miniupnp@free.fr>

#### Description

new port for GrafX2 a 256 color painting program
http://grafx2.chez.com/

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.4
Xcode 2.5

###### Verification 
Have you

- [X] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [X] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [X] checked your Portfile with `port lint`?
- [X] tried a full install with `sudo port -vst install`?
- [X] tested basic functionality of all binary files?

### CODE REVIEW

File: graphics/grafx2/Portfile [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                grafx2
+version             2.5
+categories          graphics
+maintainers         @miniupnp
```
Comment (pmetzger) [inline graphics/grafx2/Portfile]: It is best if you include both your github handle and email address (see the guide for details on the format), and if you have no specific reason to prohibit team members from making small changes without consulting you, you should set your ports "openmaintainer" as well.

File: graphics/grafx2/Portfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+long_description    GrafX2 is a bitmap paint program inspired by the Amiga programs \
+                    Deluxe Paint and Brilliance. Specialized in 256-color drawing, \
+                    it includes a very large number of tools and effects that make \
+                    it particularly suitable for pixel art, game graphics, and generally \
+                    any detailed graphics painted with a mouse.
+
+homepage            http://grafx2.chez.com/
+distname            ${name}-v${version}
+master_sites        https://gitlab.com/GrafX2/grafX2/-/archive/v${version}/::gfx2 \
+                    sourceforge:project/recoil/recoil/4.2.0:recoil
```
Comment (pmetzger) [inline graphics/grafx2/Portfile]: What is "recoil"?

File: graphics/grafx2/Portfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+distfiles           ${distname}.tar.bz2:gfx2 \
+                    recoil-4.2.0.tar.gz:recoil
+use_bzip2           yes
+checksums           ${distname}.tar.bz2 \
+                    sha256 cf634e0f1d94bc5131a4de4490d4bf07e2965d774687add2487a9fa0240aeb10 \
+                    rmd160 692e8c871ade535db741bfff730cb74bc3398327 \
```
Comment (pmetzger) [inline graphics/grafx2/Portfile]: You should probably have "size" for both of your distfiles.

File: graphics/grafx2/Portfile (hunk 2 above)
Comment (miniupnp) [reply graphics/grafx2/Portfile]: REtro COmputer Image Library http://recoil.sourceforge.net/
GrafX2 can be built without. NORECOIL=1
Maybe I should add a port variant

**Comment by kencu** [comment]
First thing I should say here is _GREAT_ job on what you did porting sdl2 to 10.4 and 10.5 PPC. THANKS for that. I have it in my TIgerPorts and LeopardPorts repo. If you ever get around to doing any later version of sdl2 than 2.03, I'll be waiting!

I installed this port on my 10.6.8 system and it went right through and worked without a hitch, so I think this is your first port submission, and good job so far.

I noticed that the build calls `gcc` as a compiler.
```
gcc -D_DARWIN_C_SOURCE -D__macosx__ -W -Wall -Wdeclaration-after-statement -O3 -std=c99 -g -I/opt/local/include/lua-5.2 -D__ENABLE_LUA__ -I/opt/local/include/SDL -D_GNU_SOURCE=1 -D_THREAD_SAFE -DNOTTF=1 -I/usr/include -I/opt/local/include/libpng16 -DENABLE_FILENAMES_ICONV  -c oldies.c -o ../obj/macosx/oldies.o
```
Although this does work, `gcc` doesn't represent a consistent compiler on MacOS -- it could be almost anything. On my system it happens to be (at present)
```
$ gcc --version
i686-apple-darwin10-gcc-4.2.1 (GCC) 4.2.1 (Apple Inc. build 5666) (dot 3)
Copyright (C) 2007 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
and that worked. We like to try to specify the compiler more fully. There is a whole writeup on this on MacPorts wiki <https://trac.macports.org/wiki/UsingTheRightCompiler>.

Whether that needs to be done or not for this port I wouldn't say just now, but it is what we generally do so that we can better spec a compiler. I'll check a bit and see.

Great job! Do more! Come back often! You're just the kind of people we like to see around here!

**Comment by miniupnp** [comment]
@kencu well I should have a try at compiling SDL-2.0.8 on my 10.4 PPC ;)

About the compiler, grafx2 should compile with any gcc or clang version... 
As there is no configure script, do I need to do something special so macports can set the right CC during build ?
I think I need to patch grafx2 to CC ?= gcc instead of plain =

**Comment by kencu** [comment]
> I think I need to patch grafx2 to CC ?= gcc instead of plain =
Yes, I believe that would do it. Then it would pick up the CC that MacPorts sends in, which is what we like, so we can peg it down better.

**Comment by kencu** [comment]
Sounds like the portion of the writeup we're interested in, in detail, is here:

<https://trac.macports.org/wiki/UsingTheRightCompiler#nonstandard-ports>

**Comment by miniupnp** [comment]
> Sounds like the portion of the writeup we're interested in, in detail,
> is here:
>
> https://trac.macports.org/wiki/UsingTheRightCompiler#nonstandard-ports
>
Thanks. It now compiles using the right CC

File: graphics/grafx2/Portfile (hunk 3 above)
Comment (miniupnp) [reply graphics/grafx2/Portfile]: done

**Comment by kencu** [comment]
beautiful job. Builds perfectly on my system, and runs as it should. Now we'll push it to the buildbots, and see what shakes loose :>

**Comment by kencu** [comment]
Ah, sorry everyone -- should have squashed these into one commit. I didn't notice they were three separate ones. Will do next time. -- Ken.

**Comment by kencu** [comment]
buildbots say we have a bit of work to do :>

<https://build.macports.org/builders/ports-10.7_x86_64_legacy-builder/builds/66850/steps/install-port/logs/stdio>

<https://build.macports.org/builders/ports-10.11_x86_64-builder/builds/55849/steps/install-port/logs/stdio>

<https://build.macports.org/builders/ports-10.12_x86_64-builder/builds/59770/steps/install-port/logs/stdio>

etc...

**Comment by miniupnp** [comment]
```
make[2]: pkg-config: Command not found
/usr/bin/clang   showimage.o  -L/opt/local/lib -lSDLmain -Wl,-framework,AppKit -lSDL -Wl,-framework,Cocoa  -o showimage
```
I guess pkg-config should be added to the build dependencies ?
should it be `depends_build   path:bin/pkg-config:pkgconfig` or `depends_build   port:pkgconfig` ?

**Comment by kencu** [comment]
I took care of that one already for you.

**Pull request merged**